plugins/pluginmanagers/lazy.nix

```diff
@@ -1,222 +1,278 @@
-{
-  lib,
-  helpers,
-  config,
-  pkgs,
-  ...
-}:
-with lib;
+{ lib, pkgs, ... }:
 let
-  cfg = config.plugins.lazy;
-  lazyPlugins = cfg.plugins;
+  inherit (lib.nixvim)
+    defaultNullOpts
+    mkNullOrOption
+    mkNullOrLuaFn
+    mkNullOrStrLuaFnOr
+    ;
+  inherit (lib) types;
+in
+lib.nixvim.neovim-plugin.mkNeovimPlugin {
+  name = "lazy";
+  originalName = "lazy.nvim";
+  package = "lazy-nvim";
+
+  maintainers = with lib.maintainers; [
+    MattSturgeon
+    my7h3le
+  ];
+
+  settingsOptions = with types; {
```

Comment:
Just a reminder: `settings` is a freeform option. This means we don't need to declare sub-options for _everything_ that can be configured.

Anything config definitions that don't match an option we've declared will use the freeformType `attrsOf anything`.

Reply:
> I can totally remove them though if it's deemed better to avoid having them 🙂

It's always a difficult judgement call. Let's revisit this towards the end of the review process, unless anyone already has a strong opinion.

Reply:
After going over everything again, I don't think there were as many settings options as I thought. With this dropped, there's currently zero.

I'm leaning towards including at least some of them :saluting_face:

Reply:
I always prefer having a decent amount of settings included to make it obvious what functionality can be customized without going to upstream docs. But, some plugins have too many that it becomes just pure tech debt to support all the settingsOptions.